### DISCUSSION TITLE
Almost amazon forms are submitting despite required on fields

### DISCUSSION THREAD

**Discussion by tsekeres** [nss-evening-cohort-14/evening-client-side#153] [Help Me!] [answered]
Your Name:
TAD
Have you searched the other help tickets?Affirmative

Describe your issue:
After adding the create authors functions and such on Saturday, I noticed that both the add books form and the add author form will both add a blank card if the submit button is pushed. I don't know if they did that before the changes. I did see that Dr T's required designation was working before she added the e.prevent.default in class. 

List any error messages that appear. NONE

What have you googled?
https://www.w3schools.com/tags/att_input_required.asp
https://stackoverflow.com/questions/9469994/submit-button-checking-mandatory-fields-before-opening-new-window
https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation

What have you tried? 
1. Commenting out the required designation.
2. Commenting out the e.prevent.defense.
3. Tracing the paths of buttons/eventListeners/form builders/create functions for errors.

Copy/paste link TO THE FILE on your branch or in your repo that you are having issues with:
https://github.com/tsekeres/TS-almost-amazon/tree/create-authors/src/javascripts

**Comment by drteresavasquez** [comment]
If you want to add validation you will need to do so using JavaScript. I spoke about this in class and said that by using prevent default the default behaviors for the form would not function including required on the inputs. This is not a focus for this project but if you want to challenge yourself, you are free to research adding form validation with JS.

**Reply by tsekeres** [reply]
OK, Thanks. So, I didn't break it. Good to know.

But it also did not fix it to take away the prevent default. So I was confused. 

I'll research later.

**Comment by tsekeres** [comment, marked as answer]
Ok Thanks! Not broken, just conflicting code that cancels required designation.
